python/pyarrow/tests/test_udf.py

```diff
@@ -504,3 +504,132 @@ def test_input_lifetime(unary_func_fixture):
     # Calling a UDF should not have kept `v` alive longer than required
     v = None
     assert proxy_pool.bytes_allocated() == 0
+
+
+def test_aggregate_udf_with_custom_state():
+    class State:
+        def __init__(self, non_null=0):
+            self._non_null = non_null
+
+        @property
+        def non_null(self):
+            return self._non_null
+
+        @non_null.setter
+        def non_null(self, value):
+            self._non_null = value
```

Comment:
I'm not much of a python expert but getters and setters seem like overkill here.  Are they needed?